packages/block-library/src/post-featured-image/index.php

```diff
@@ -54,9 +54,26 @@ function render_block_core_post_featured_image( $attributes, $content, $block )
 	}
 
 	$featured_image = get_the_post_thumbnail( $post_ID, $size_slug, $attr );
+
+	// Get the first image from the post.
+	if ( $attributes['useFirstImageFromPost'] && ! $featured_image ) {
+		$content_post = get_post( $post_ID );
+		$content = $content_post->post_content;
+		$processor = new WP_HTML_Tag_Processor( $content );
```

Comment:
given the peculiar use of the Tag Processor here I think it warrants an explanatory comment. something indicating what the goal is.

```php
/*
 * Transfer the image tag from the post into a new text snippet.
 * Because the HTML API doesn't currently expose a way to extract
 * HTML substrings this is necessary as a workaround. Of note, this
 * is different than directly extracting the IMG tag:
 *  - If there are duplicate attributes in the source there will only be one in the output.
 *  - If there are single-quoted or unquoted attributes they will be double-quoted in the output.
 *  - If there are named character references in the attribute values they may be replaced with their direct code points. E.g. `&hellip;` becomes `…`.
 *
 * In the future there will likely be a mechanism to copy snippets of HTML from
 * one document into another, via the HTML Processor's `get_outer_html()` or
 * equivalent. When that happens it would be appropriate to replace this custom
 * code with that canonical code.
 */
```

For one off projects I wouldn't be so worried, but I'd like all Core code to be clear about what _is_ exemplary and _should_ be copied vs. what is _not_ meant to serve as a canonical example of use.

Reply:
Done in [a161d4d](https://github.com/WordPress/gutenberg/pull/56573/commits/a161d4dcfdf937f023453d16245086b9fc97dd06)